### PR TITLE
Move away from transparency and towards verifiability as a property

### PR DESCRIPTION
**Pull request by chris-wood**
As discussed someone indirectly on the list, it might make sense to accommodate different types of transparency services, in particular those that don't require running and operating a publicly-verifiable ledger. To that end, this change moves away from _transparency_ as a requirement for SCITT and focuses on _verifiability_, allowing for SCITT instances to implement verifiability _with_ transparency (as in CT) if desired, but also allowing for simpler deployments (like those based on a timestamping service) where applicable.

### CODE REVIEW

File: draft-birkholz-scitt-architecture.md
Comment (thomas-fossati) [inline draft-birkholz-scitt-architecture.md]:
```suggestion
  I-D.birkholz-scitt-receipts: RECEIPTS
```

Comment (henkbirkholz) [reply draft-birkholz-scitt-architecture.md]: oh

**Review by henkbirkholz — APPROVED** [review]

File: draft-birkholz-scitt-architecture.md
Comment (joshuagl) [inline draft-birkholz-scitt-architecture.md]: This seems like an unintended typo?

File: draft-birkholz-scitt-architecture.md
Comment (ad-l) [inline draft-birkholz-scitt-architecture.md]: Making the ledger optional is a significant weakening of the technical guarantee of SCITT, which is principally defined based on the ledger. We think that most of the costs associated with the implementation and operation of SCITT servers are related to the **persistence and distribution/querying of the ledger** rather than its actual computation. For instance, it is possible to implement a minimal append-only ledger using a single hash (like PCRs in TPMs), which technically meets the requirements of append-only and consistent/fork-free. The implementation overhead for such a ledger is only the computation of an extra hash per claim and the persistence of a single hash value in addition to the receipt signing key, while the benefit is a much stronger a clearly defined security goal for SCITT verifiers. I would like to check there is consensus for this change before it is approved.

File: draft-birkholz-scitt-architecture.md
Comment (ad-l) [inline draft-birkholz-scitt-architecture.md]: (since the ledger is optional, appending the claim to the ledger is also optional)

File: draft-birkholz-scitt-architecture.md
Comment (ad-l) [inline draft-birkholz-scitt-architecture.md]: The security property is very underspecified. As written, the baseline guarantee is that the registration policy was satisfied when the claim was registered. Additional guarantees stemming from the addition of the claim to the ledger depend on the use of a ledger, the type of ledger/receipt, and how the server maintains and distributes the ledger.

File: draft-birkholz-scitt-architecture.md
Comment (ad-l) [inline draft-birkholz-scitt-architecture.md]: The definition of ledger is no longer valid when the ledger is optional. A receipt is now a witness that a claim has been registered on SCITT service, with optional universally verifiable proof that the claim has been appended to the ledger.

**Review by ad-l — CHANGES_REQUESTED** [review]

**Review by OR13 — APPROVED** [review]

**Comment by OR13** [comment]
I wonder if we can get a clearer change request, what actions need to be performed to obtain approval.

perhaps we can open issues and merge as is?

File: draft-birkholz-scitt-architecture.md
Comment (henkbirkholz) [reply draft-birkholz-scitt-architecture.md]: I agree with @ad-l
Do others share Chris' concern?

**Comment by henkbirkholz** [comment]
@OR13 , I am under the assumption that making the ledger optional would cut away a significant core concept. Not sure how to accommodate Chris' ask. @chris-wood, could you reply to @ad-l feedback, so we can resolve the discussion?

**Comment by OR13** [comment]
> it might make sense to accommodate different types of transparency services, in particular those that don't require running and operating a publicly-verifiable ledger.

My understanding is that the architecture has already incorporated this feedback.

In my latest tests I showed how to do this end to end with a private ledger.

I agree with the motivation / framing, but the specific word changes don't achieve the objective imo.

The ledger isn't `optional` in the architecture, but it does not need to be "public access"... and it might now always be needed is a use case / story.

There are a lot of changes in this PR, and its got a lot of outdated comments, making it hard to review easily.

After looking at the changes directly, I think they can be accepted as is.

File: draft-birkholz-scitt-architecture.md
Comment (OR13) [inline draft-birkholz-scitt-architecture.md:91]:
```suggestion
The first guarantee is achieved by requiring issuers to sign their statements and associated metadata using a distributed public key infrastructure. The second guarantee is achieved by appending the signed statement to a data structure that logically defines an immutable, append-only, transparent registry for the Registration Service (RS). The last guarantee is achieved by using a verifiable data structure to implement the registry (such as a Merkle Tree), and by requiring a Registration Service that operates the registry to endorse its state at the time of registration.
```

**Comment by chris-wood** [comment]
> My understanding is that the architecture has already incorporated this feedback.

Indeed -- I think @ad-l already incorporated a better version of this change into the architecture document (though I've not read the latest version). This can be closed.

**Pull request closed without merging**
